### ISSUE TITLE
Warning: UCX is unable to handle VM_UNMAP

### ISSUE THREAD

**Issue by devreal** [open-mpi/ompi#9859]
I see below error on an IB cluster with OMPI 5.0.x and UCX 1.11.0:

```
[r39c3t4n4:108810] ../../../../../opal/mca/common/ucx/common_ucx.c:390  Warning: UCX is unable to handle VM_UNMAP event. This may cause performance degradation or data corruption. Pls try adding --mca opal_common_ucx_opal_mem_hooks 1 to mpirun/oshrun command line to resolve this issue.
```

@gpaulsen pointed me to https://github.com/openucx/ucx/issues/4866 which suggested that disabling pml/yalla in OMPI was a solution. However, pml/yalla has been removed in OMPI 5.0.x.

If I disable either btl/uct or pml/ucx the warning disappears. Disabling any other installed pml (ob1, cm, or v in my case) or btl (tcp, sm, self) does not help. I suspect that there is some interference between btl/uct and pml/ucx.

The warning is printed on every process, which is pretty annoying in larger runs. It is also not clear what performance impact the solution of setting `--mca opal_common_ucx_opal_mem_hooks 1` would have. Can there be no automatic fall-back?

**Comment by devreal** [comment]
@bwbarrett Suggested to try building OMPI with `--enable-mca-no-build=memory-patcher,memory-malloc_solaris`. In this case, the warnings disappear as well.

**Comment by gpaulsen** [comment]
Talked to @janjust today, he'll take a look.

**Comment by janjust** [comment]
@devreal looks like it got introduced with `0909e1448d UCT: require at least UCX 1.9.0 ` #9660 

So seems to be a conflict between btl/uct and pml/ucx

**Comment by devreal** [comment]
So do we have to disable btl/uct if pml/ucx is used? In that case we would not be able to use osc/rdma with btl/uct while pml/ucx is used...

**Comment by janjust** [comment]
yes, I see the dilemma, let me think about it some more - I don't think we should force the exclusion

**Comment by janjust** [comment]
incidentally, this has a huge impact on ucx performance

**Comment by abouteiller** [comment]
There is the exact same issue in v4.1.2 with a build that contains `pml_ucx` and `btl_openib`. 

To add more use cases as to why disabling one of the two is problematic, we have the same system image on nodes with mlx5, mlx4 hardware, that use pml_ucx, and mtha nodes that use openib. Other than this incompatibility the automatic selection between the two works great and lets us have a single configuration.

**Comment by hjelmn** [comment]
How about ```disable_ucx_memory_hooks=false```? We added that specifically because UCX incompatibilities with Open MPI memory hooks. Should probably be linked to the common/ucx version of the variable.

**Comment by abouteiller** [comment]
`disable_ucx_memory_hooks` applies only to `uct`, the issue is more pervasive as it also happens with `openib`. 

playing with more parameters, I note that `pml_ucx_opal_mem_hooks` and `osc_ucx_opal_mem_hooks` pretend they are synonym with `opal_common_ucx_opal_mem_hooks` but setting the `osc` one alone doesn't resolve the issue, so it clearly isn't a full synonym. 

On my test systems, performance appears to be identical between `pml=ucx btl=self` and `pml=ucx btl=openib,self common_ucx_opal_mem_hooks=1`, is there a rationale for why the later defaults to 0 rather than 1?

**Comment by brminich** [comment]
@hoopoepg

**Comment by hoopoepg** [comment]
@brminich will look

**Comment by hoopoepg** [comment]
@devreal could you check this PR https://github.com/open-mpi/ompi/pull/9946

thank you
